coreos-kubernetes-multimaster/azuredeploy.json

```diff
@@ -0,0 +1,832 @@
+        "accountType": "[variables('adminStorageAccountType')]"
+      }
+    },
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('workerStorageAccountName')]",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2015-06-15",
+      "properties": {
+        "accountType": "[variables('workerStorageAccountType')]"
+      }
+    },
+    {
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "[concat(variables('etcdVmNamePrefix'), copyindex())]",
+      "copy": {
+        "name": "etcdVmLoop",
+        "count": "[parameters('etcdNumberOfNodes')]"
+      },
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "etcdNicLoop",
+        "[concat('Microsoft.Storage/storageAccounts/', variables('adminStorageAccountName'))]",
+        "[concat('Microsoft.Compute/availabilitySets/', variables('etcdAvailabilitySetName'))]"
+      ],
+      "apiVersion": "2015-06-15",
+      "properties": {
+        "availabilitySet": {
+          "id": "[resourceId('Microsoft.Compute/availabilitySets',variables('etcdAvailabilitySetName'))]"
+        },
+        "hardwareProfile": {
+          "vmSize": "[parameters('etcdVmSize')]"
+        },
+        "osProfile": {
+          "computerName": "[concat(variables('etcdVmNamePrefix'), copyindex())]",
+          "adminUsername": "[parameters('adminUsername')]",
+          "customData": "[base64(concat('#cloud-config\n\ncoreos:\n  update:\n    reboot-strategy: \"etcd-lock\"\n  etcd2:\n    name: ',concat('etcd', copyIndex()),'\n    initial-cluster: ', variables('etcdConnectionStringBinary'),'\n    advertise-client-urls: http://$private_ipv4:2379,http://$private_ipv4:4001\n    initial-advertise-peer-urls: http://$private_ipv4:2380\n    listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001\n    listen-peer-urls: http://0.0.0.0:2380\n    initial-cluster-token: etcd-kube\n    heartbeat-interval: 250\n    election-timeout: 1250\n    initial-cluster-state: new\n  units:\n    - name: etcd2.service\n      command: start\n    - name: fleet.service\n      command: start'))]",
+          "linuxConfiguration": {
+            "disablePasswordAuthentication": "true",
+            "ssh": {
+              "publicKeys": [
+                {
+                  "path": "[variables('sshKeyPath')]",
+                  "keyData": "[parameters('sshKeyData')]"
+                }
+              ]
+            }
+          }
+        },
+        "storageProfile": {
+          "imageReference": {
+            "publisher": "CoreOS",
+            "offer": "CoreOS",
+            "sku": "[parameters('imageSku')]",
+            "version": "latest"
+          },
+          "osDisk": {
+            "name": "[concat(variables('etcdVmNamePrefix'), copyindex())]",
+            "vhd": {
+              "uri": "[concat('http://',variables('adminStorageAccountName'),'.blob.core.windows.net/vhds/',variables('etcdVmNamePrefix'), copyindex(),'.vhd')]"
```

Comment:
Please see the "all clouds" entry in the best practices checklist: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md